wkcuber/wkcuber/_internal/utils.py

```diff
@@ -65,6 +68,34 @@ def parse_padding(padding_str: str) -> Tuple[int, ...]:
         raise argparse.ArgumentTypeError("The padding could not be parsed") from e
 
 
+class AsciiArgumentParser(ArgumentParser):
+    """
+    This class is a lightweight wrapper around ArgumentParser and checks
+    that the shell command doesn't contain non-ascii characters since these
+    can easily disturb the argument parsing.
+
+    If non-ascii characters are found, an error is raised. This behavior
+    can be disabled by passing --allow-non-ascii on the command line.
+    """
+
+    IGNORE_FLAG = "--allow-non-ascii"
+
+    def parse_args(self, args: Optional[Sequence[str]] = None, namespace: Optional[Namespace] = None) -> Any:  # type: ignore
+        if args is None:
+            args = sys.argv[1:]
+
+        results = re.findall(r"[^\x00-\x7F]", " ".join(args))
```

Comment:
Do you mind adding a comment to the RegEx. I have a hard time understanding the character range it is searching for. 

I wonder if it would be easier to express the condition as `ascii_code > 128`, e..g `containsNonAscii = any([ord(el) > 128 for el in sys.argv])`?